cmd/maya-nodebot/mcrd/controllerspc.go

```diff
@@ -0,0 +1,369 @@
+	// You can use DeepCopy() to make a deep copy of original object and modify this copy
+	// Or create a copy manually for better performance
+	spcCopy := spc.DeepCopy()
+	//spcCopy.Status.AvailableReplicas = 1
+	// Until #38113 is merged, we must use Update instead of UpdateStatus to
+	// update the Status block of the spc resource. UpdateStatus will not
+	// allow changes to the Spec of the resource, which is ideal for ensuring
+	// nothing other than resource status has been updated.
+	_, err := c.clientset.OpenebsV1alpha1().StoragePoolClaims(spc.Namespace).Update(spcCopy)
+	return err
+}
+
+// enqueueSpc takes a spc resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than spc.
+func (c *Controller) enqueueSpcAdd(obj interface{}) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	c.workqueue.AddRateLimited(key)
+
+	c.handleObjectAdd(obj)
+}
+
+func (c *Controller) enqueueSpcUpdate(obj interface{}) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	c.workqueue.AddRateLimited(key)
+	//printing since the update corner case is yet to be decided
+	fmt.Println("key updated: ", key)
+	//fmt.Println("obj updated: ", obj)
+}
+
+func (c *Controller) enqueueSpcDelete(obj interface{}) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	c.workqueue.AddRateLimited(key)
+
+	//simply printing the deleted key
+	fmt.Println("key deleted: ", key)
+	//fmt.Println("obj deleted: ", obj)
+}
+
+// handleObject will take any resource implementing metav1.Object and attempt
+// to find the spc resource that 'owns' it. It does this by looking at the
+// objects metadata.ownerReferences field for an appropriate OwnerReference.
+// It then enqueues that spc resource to be processed. If the object does not
+// have an appropriate OwnerReference, it will simply be skipped.
+func (c *Controller) handleObjectAdd(obj interface{}) {
+	var object metav1.Object
+	var ok bool
+	if object, ok = obj.(metav1.Object); !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			runtime.HandleError(fmt.Errorf("error decoding object, invalid type"))
+			return
+		}
+		object, ok = tombstone.Obj.(metav1.Object)
+		if !ok {
+			runtime.HandleError(fmt.Errorf("error decoding object tombstone, invalid type"))
+			return
+		}
+		glog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
+	}
+	glog.V(4).Infof("Processing object: %s", object.GetName())
+
+	spc, err := c.clientset.OpenebsV1alpha1().
+		StoragePoolClaims(object.GetNamespace()).
+		Get(object.GetName(), metav1.GetOptions{})
+	if err != nil {
+		fmt.Println("error", err)
```

Comment:
Why can we not return errors ?

Reply:
I will change it when we implement daemonset, I will log them. Right now, I was printing both outputs/errors on the console.